LEAF_Request_Portal/js/formSearchMultisite.js

```diff
@@ -0,0 +1,1111 @@
+        if (typeof advSearch[i].match == "string") {
+          $("#" + prefixID + "widgetMat_" + i).val(
+            advSearch[i].match.replace(/\*/g, "")
+          );
+        }
+      }
+    }
+  }
+
+  /**
+   * @memberOf LeafFormSearchMultisite
+   * From: http://werxltd.com/wp/2010/05/13/javascript-implementation-of-javas-string-hashcode-method/
+   */
+  function getLocalStorageHash() {
+    var hash = 0,
+      i,
+      chr,
+      len;
+    if (document.URL.length == 0) return hash;
+    for (i = 0, len = document.URL.length; i < len; i++) {
+      chr = document.URL.charCodeAt(i);
+      hash = (hash << 5) - hash + chr;
+      hash |= 0; // Convert to 32bit integer
+    }
+    return hash;
+  }
+
+  /**
+   * @memberOf LeafFormSearchMultisite
+   */
+  function setOrgchartPath(path) {
+    orgchartPath = path;
+  }
+
+  /**
+   * @memberOf LeafFormSearchMultisite
+   */
+  function getLastSearch() {
+    return localStorage.getItem(localStorageNamespace + ".search");
+  }
+
+  /**
+   * @memberOf LeafFormSearchMultisite
+   */
+  function setSearchFunc(func) {
+    searchFunc = func;
+  }
+
+  /**
+   * @memberOf LeafFormSearchMultisite
+   */
+  function search(txt) {
+    if (txt != "*") {
+      localStorage.setItem(localStorageNamespace + ".search", txt);
+    }
+    return searchFunc(txt);
+  }
+
+  /**
+   * @memberOf LeafFormSearchMultisite
+   */
+  function inputLoop() {
+    if ($("#" + prefixID + "searchtxt") == null) {
+      clearInterval(intervalID);
+      return false;
+    }
+    timer += timer > 5000 ? 0 : 200;
+    if (timer > 400) {
+      var txt = $("#" + prefixID + "searchtxt").val();
+
+      if (txt != "" && txt != q) {
+        q = txt;
+
+        if (
+          currRequest != null &&
+          currRequest.abort != undefined &&
+          typeof currRequest.abort == "function"
+        ) {
+          currRequest.abort();
+        }
+
+        currRequest = search(txt);
+      } else if (txt == "") {
+        if (txt != q) {
+          search("");
+        }
+        q = txt;
+        $("#" + this.prefixID + "_result").html("");
+        numResults = 0;
+        showNotBusy();
+      } else {
+        showNotBusy();
+      }
+    }
+  }
+
+  /**
+   * @memberOf LeafFormSearchMultisite
+   */
+  function focus() {
+    $("#" + prefixID + "searchtxt").focus();
+  }
+
+  /**
+   * @memberOf LeafFormSearchMultisite
+   */
+  function showBusy() {
+    $("#" + prefixID + "searchIcon").css("display", "none");
+    $("#" + prefixID + "searchIconBusy").css("display", "inline");
+    $(".status").text("Loading");
+  }
+
+  /**
+   * @memberOf LeafFormSearchMultisite
+   */
+  function showNotBusy() {
+    $("#" + prefixID + "searchIcon").css("display", "inline");
+    $("#" + prefixID + "searchIconBusy").css("display", "none");
+  }
+
+  /**
+   * @memberOf LeafFormSearchMultisite
+   */
+  function createEmployeeSelectorWidget(widgetID, type) {
+    if (type == undefined) {
```

Comment:
Do we want this to be typeof like below?

Reply:
While both are valid, I like the former better since it describes what we're doing more accurately: checking to see if the variable is undefined rather than checking that the type is undefined.

I went down a rabbit hole for this one... apparently "typeof" was preferred (in jQuery) for Global variables in the past because "undefined" used to be a mutable variable in older browsers.